src/app/ng2-ui-auth/lib/ng2-ui-auth.module.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ *  Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+import { NgModule, ModuleWithProviders } from '@angular/core';
+import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
+import { IConfigOptions, IPartialConfigOptions } from './config-interfaces';
+import { CONFIG_OPTIONS, ConfigService } from './config.service';
+import { StorageService } from './storage-service';
+import { BrowserStorageService } from './browser-storage.service';
+import { SharedService } from './shared.service';
+import { JwtInterceptor } from './interceptor.service';
+import { OauthService } from './oauth.service';
+import { HttpClient } from '@angular/common/http';
+import { PopupService } from './popup.service';
+import { LocalService } from './local.service';
+import { AuthService } from './auth.service';
+
+@NgModule({
+  imports: [HttpClientModule],
+  declarations: [],
+  exports: [],
+})
+export class Ng2UiAuthModule {
+  static forRoot(configOptions?: IPartialConfigOptions, defaultJwtInterceptor = true): ModuleWithProviders<Ng2UiAuthModule> {
```

Comment:
This was the main change, adding the type to `ModuleWithProviders<Ng2UiAuthModule>`.